src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrOfObjects = [];
+  let copyState = {};
+  let objectOfEveryProperty = {};
+
+  Object.assign(copyState, state);
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        objectOfEveryProperty = Object.assign({}, copyState, obj.extraData);
+        arrOfObjects.push(objectOfEveryProperty);
+        copyState = {};
+        Object.assign(copyState, objectOfEveryProperty);
+        break;
+      case 'removeProperties':
+        obj.keysToRemove.forEach(elem => delete copyState[elem]);
+        objectOfEveryProperty = Object.assign({}, copyState);
+        copyState = {};
+        Object.assign(copyState, objectOfEveryProperty);
+        arrOfObjects.push(objectOfEveryProperty);
+        break;
+
+      case 'clear':
+        for (const elem in copyState) {
+          delete copyState[elem];
+        }
+        objectOfEveryProperty = Object.assign({}, copyState);
+        arrOfObjects.push(objectOfEveryProperty);
+        break;
```

Comment:
Too much copy-paste in cases, check [checklist](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) pt.4